core/modules/wdb/QueryRunner.cc

```diff
@@ -459,27 +459,25 @@ bool QueryRunner::_dispatchChannel() {
     size_t tSize = 0;
 
     try {
-        if (_cancelled) {
-            break;
-        }
-
-        proto::TaskMsg_Fragment const& fragment(m.fragment(fragNum));
-        string const query = task->getQueryStr();
-
-        util::Timer sqlTimer;
-        sqlTimer.start();
-        MYSQL_RES* res = _primeResult(query); // This runs the SQL query, throws SqlErrorObj on failure.
-        sqlTimer.stop();
-        LOGS(_log, LOG_LVL_DEBUG, " fragment time=" << sqlTimer.getElapsed() << " query=" << query);
-        _fillSchema(res);
-        numFields = mysql_num_fields(res);
-        // TODO fritzm: revisit this error strategy
-        // (see pull-request for DM-216)
-        // Now get rows...
-        if (!_fillRows(res, numFields, rowCount, tSize)) {
-            erred = true;
+        if (!_cancelled) {
+            //&&& proto::TaskMsg_Fragment const& fragment(m.fragment(fragNum));
+            string const& query = _task->getQueryString();
```

Comment:
Is it safe to keep a reference to an internal string of a tasks? Tasks may disappear from queries for other reasons and result in mysterious crashes that will require debugging.

Reply:
The _task object will be good until execution leaves QueryRunner::runQuery(). Everything else is called by runQuery().